x/authenticator/ante/authenticator_test.go

```diff
@@ -0,0 +1,323 @@
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	feeCoins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	err := s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	s.Require().NoError(err)
+
+	tx, _ := GenTx(
+		s.EncodingConfig.TxConfig,
+		[]sdk.Msg{
+			testMsg1,
+			testMsg2,
+		},
+		feeCoins,
+		300000,
+		"",
+		[]uint64{0, 0},
+		[]uint64{0, 0},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+	)
+
+	anteHandler := sdk.ChainAnteDecorators(s.AuthenticatorDecorator)
+	_, err = anteHandler(s.Ctx, tx, false)
+
+	s.Require().NoError(err)
+}
+
+// TestSignatureVerificationWithAuthenticatorInStore test out of gas error
+func (s *AutherticatorAnteSuite) TestSignatureVerificationOutOfGas() {
+	osmoToken := "osmo"
+	coins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	// Create a test messages for signing
+	testMsg1 := &banktypes.MsgSend{
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[0]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	testMsg2 := &banktypes.MsgSend{
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	feeCoins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	err := s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[1].PubKey().Bytes(),
+	)
+	s.Require().NoError(err)
+
+	tx, _ := GenTx(
+		s.EncodingConfig.TxConfig,
+		[]sdk.Msg{
+			testMsg1,
+			testMsg2,
+		},
+		feeCoins,
+		300000,
+		"",
+		[]uint64{0, 0},
+		[]uint64{0, 0},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[1],
+			s.TestPrivKeys[1],
+		},
+	)
+
+	anteHandler := sdk.ChainAnteDecorators(s.AuthenticatorDecorator)
+	_, err = anteHandler(s.Ctx, tx, false)
+
+	// TODO: improve this test for gas consumption
+	fmt.Println("Gas Consumed: after txn gas over 20000")
+	fmt.Println(s.Ctx.GasMeter().GasConsumed())
+
+	s.Require().Error(err)
```

Comment:
I think here we can test at least that the test contains the word gas:

```suggestion
	s.Require().Error(err)
	s.Require().ErrorContains(err, "gas")
```